scripts/telegram_vc_adapter.py

```
@@ -51,38 +65,106 @@ def status(self) -> dict[str, Any]:
+    def join(self, chat_id: str, source: str, camera: bool = True) -> dict[str, Any]:
         if not source.strip():
-            raise RuntimeError("Choose an RTMP URL or media source before going live")
+            raise AdapterError("Choose an RTMP URL or media source before going live")
         self.ensure_client()
         numeric_chat_id = int(chat_id)
-        self.calls.play(numeric_chat_id, source)
+        if camera:
+            try:
+                from pytgcalls.types import AudioVideoPiped
+                self.calls.play(numeric_chat_id, AudioVideoPiped(source))
+            except Exception:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)